### PR TITLE
CI: remove coverage jobs

### PR DESCRIPTION
**Pull request by tupui**
Part of https://github.com/scipy/scipy/issues/15814#issuecomment-1506516709

Remove all coverage related jobs.

Shall I add coverage to a GH job? Without the upload to codecov, just running with coverage on?

### CODE REVIEW

**Comment by andyfaff** [comment]
This PR is going to overlap with moving the prerelease job, so I don't think it makes sense to merge this just yet. I'd like to get @tylerjereddy POV on coverage, he sometimes comments on that aspect of PRs.

**Comment by rgommers** [comment]
> Shall I add coverage to a GH job? Without the upload to codecov, just running with coverage on?

**Comment by tupui** [comment]
ok thanks, I will update the PR now that we merged the pre-release one.

**Review by tylerjereddy — APPROVED** [review]
The diff looks good to me. I'm indeed a bit sad to see the automated CI entry with the coverage % disappear, but I suspect we may get something like it again someday. I've had good experiences interacting with Tom Hu at codecov, but sounds like we've just had too many maintenance concerns and maybe not that much usage of the reports anyway.



I agree with Ralf that it may not hurt to have a `gcov`-enabled job in GHA. I think that makes the tests run slower b/c of tracing, but making sure it still works is useful to ensure that we can check the coverage locally for i.e., really big PRs.

From a quick local check, I think the reviewer burden for checking coverage data locally is probably lower with the new build system we have. Running a clean build *and* full test suite takes me about `3:49` on an older desktop and `2:46` (!!) on a newer laptop, so for big PRs it isn't too unreasonable for the reviewer to generate a report locally I suppose.

**Comment by tupui** [comment]
I just added coverage and gcov to the pre-release job which just got moved to actions. Good thing is that it still had the old name with coverage 😅 

By the way, now that I am looking at the file `linux_meson.yml`, we have a job which is checking deps from their `main` and a job which is using pre-releases. Do we need both??

**Comment by tupui** [comment]
Ark `gcov` is a build thing, will fix that.

**Comment by rgommers** [comment]
> By the way, now that I am looking at the file `linux_meson.yml`, we have a job which is checking deps from their `main` and a job which is using pre-releases. Do we need both??

Not the same packages, the pre-releases one is a lot broader. We don't really want `matplotlib`'s main branch, but for some key dependencies we do. May need a small tweak, but it's not far off. Let's get through the Azure removal first I'd suggest.

**Comment by tupui** [comment]
Grrr re-adding the pin for pytest-cov. The bug is fixed but they need to make a release, I was too quick here.

**Comment by tupui** [comment]
Actually what about not using `pytest-cov` but plain `coverage`? I testing this locally and it works just fine:

```
coverage run dev.py test --tests scipy/stats/tests/test_survival.py
coverage report
```

File: .github/workflows/linux_meson.yml [hunk 1]
```diff
@@ -292,9 +292,9 @@ jobs:
         #       have been ironed out.
         python -m pip install ninja meson meson-python wheel click rich_click pydevtool "cython==0.29.33"
         python -m pip install --pre --upgrade --timeout=60 -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
-        # Use pytest-xdist 2.5.0 due to https://github.com/pytest-dev/pytest-cov/issues/557
-        # can update once pytest-cov>4.0
-        python -m pip install --pre --upgrade pytest pytest-cov==2.5.0 pytest-xdist pybind11 mpmath gmpy2 pythran threadpoolctl pooch matplotlib
+        # Use pytest-xdist<4.0 due to https://github.com/pytest-dev/pytest-cov/issues/557
+        # can update once pytest-cov>4.0 and remove warning filtering in pytest.ini
```
Comment (andyfaff) [inline .github/workflows/linux_meson.yml:296]: which warning filter?

**Comment by tupui** [comment]
Ok locally I could fix the mess. Let's see... Still running coverage only makes sense if we store the artifact. Let me know if I should do it. Otherwise we can just remove the install of coverage and the option. The fix in Pytest's config should stay as it's useful if someone is running this locally.

File: .github/workflows/linux_meson.yml (hunk 1 above)
Comment (tupui) [reply .github/workflows/linux_meson.yml:296]: See the newly modified file `pytest.ini` (I will push a new comment for the missing newline)

**Comment by andyfaff** [comment]
If we're going to run coverage it makes sense to see some output somewhere.

Is there a quick summary that could be produced from a coverage run that we could see printed out? That would be easier than uploading.

**Comment by tylerjereddy** [comment]
You can certainly print output to the terminal yeah

File: .github/workflows/linux_meson.yml [hunk 2]
```diff
@@ -326,4 +328,4 @@ jobs:
         export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
 
         export OPENBLAS_NUM_THREADS=1
-        python dev.py test -j2 --mode full
+        python dev.py test -j2 --mode full --coverage
```
Comment (tupui) [inline .github/workflows/linux_meson.yml:331]:
```suggestion
        python dev.py test -j2 --mode full
```

**Review by tupui — COMMENTED** [review]
Putting here the changes needed to remove `--coverage` if we are ok to leave this for a follow up (I can see to do it this week even.) and not block this more.

**Comment by tupui** [comment]
Yay seems to be working fine now!

So it's adding a chunk (doubling the log size), but I think it's still readable. What do you think??

**Comment by tylerjereddy** [comment]
I read the diff over again, checked the CI log output, and merged it, thanks.

**Pull request merged**